### ISSUE TITLE
Not honouring `max.poll.interval.ms` when log is enabled `go.logs.channel.enable`

### ISSUE THREAD

**Issue by slice-aayushs** [confluentinc/confluent-kafka-go#934]
Description
===========
When enabling Kafka logs via channels, the property `max.poll.interval.ms` is not honored. The consumer doesn't leave the group even if it hasn't polled in `max.poll.interval.ms`

```go
"go.logs.channel.enable": true,
```

How to reproduce
================

## Code With Log Channel Disabled - Honoring `max.poll.interval.ms`

```go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"
	"time"

	"github.com/confluentinc/confluent-kafka-go/kafka"
)

func main() {
	bootstrapServers := "xxx-xxxxx.ap-south-1.aws.confluent.cloud:9092"
	username := "xxxxxxxx"
	password := "xxxxxx"
	topics := []string{"topic"}
	sigchan := make(chan os.Signal, 1)
	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)

	c, err := kafka.NewConsumer(&kafka.ConfigMap{
		"bootstrap.servers":      bootstrapServers,
		"group.id":               "group",
		"enable.auto.commit":     false,
		"broker.address.family":  "v4",
		"security.protocol":      "SASL_SSL",
		"sasl.mechanism":         "PLAIN",
		"sasl.username":          username,
		"sasl.password":          password,
		"heartbeat.interval.ms":  100,
		"max.poll.interval.ms":   6000,
		"session.timeout.ms":     6000,
		"go.logs.channel.enable": false,
		"auto.offset.reset":      "earliest"})

	if err != nil {
		fmt.Fprintf(os.Stderr, "Failed to create consumer: %s\n", err)
		os.Exit(1)
	}

	fmt.Printf("Created Consumer %v\n", c)

	if err != nil {
		fmt.Printf("Failed to create schema registry client: %s\n", err)
		os.Exit(1)
	}

	if err != nil {
		fmt.Printf("Failed to create deserializer: %s\n", err)
		os.Exit(1)
	}

	err = c.SubscribeTopics(topics, nil)
	if err != nil {
		panic(err)
	}

	go func() {
		for ev := range c.Logs() {
			fmt.Println("Kafka Logs: ", ev.String())
		}
	}()

	run := true

	for run {
		select {
		case sig := <-sigchan:
			fmt.Printf("Caught signal %v: terminating\n", sig)
			run = false
		default:
			ev := c.Poll(100)
			if ev == nil {
				continue
			}

			switch e := ev.(type) {
			case *kafka.Message:
				fmt.Println("Message Received")
				time.Sleep(10 * time.Second)
			case kafka.Error:
				// Errors should generally be considered
				// informational, the client will try to
				// automatically recover.
				fmt.Fprintf(os.Stderr, "%% Error: %v: %v\n", e.Code(), e)
			default:
				fmt.Printf("Ignored %v\n", e)
			}
		}
	}

	fmt.Printf("Closing consumer\n")
	c.Close()
}
```

Logs With Log Channel Disabled
```
Created Consumer rdkafka#consumer-1
Message Received
%4|1674458805.901|MAXPOLL|rdkafka#consumer-1| [thrd:main]: Application maximum poll interval (6000ms) exceeded by 253ms (adjust max.poll.interval.ms for long-running message processing): leaving group
% Error: Local: Maximum application poll interval (max.poll.interval.ms) exceeded: Application maximum poll interval (6000ms) exceeded by 253ms
Message Received
%4|1674458816.326|MAXPOLL|rdkafka#consumer-1| [thrd:main]: Application maximum poll interval (6000ms) exceeded by 430ms (adjust max.poll.interval.ms for long-running message processing): leaving group
% Error: Local: Maximum application poll interval (max.poll.interval.ms) exceeded: Application maximum poll interval (6000ms) exceeded by 430ms
^CCaught signal interrupt: terminating
Closing consumer
```

## Code with Log channel Enabled - Not Honoring `max.poll.interval.ms`

```go
package main

import (
	"fmt"
	"os"
	"os/signal"
	"syscall"
	"time"

	"github.com/confluentinc/confluent-kafka-go/kafka"
)

func main() {
	bootstrapServers := "xxx-xxxxx.ap-south-1.aws.confluent.cloud:9092"
	username := "xxxxxxxx"
	password := "xxxxxx"
	topics := []string{"topic"}
	sigchan := make(chan os.Signal, 1)
	signal.Notify(sigchan, syscall.SIGINT, syscall.SIGTERM)

	c, err := kafka.NewConsumer(&kafka.ConfigMap{
		"bootstrap.servers":      bootstrapServers,
		"group.id":               "group",
		"enable.auto.commit":     false,
		"broker.address.family":  "v4",
		"security.protocol":      "SASL_SSL",
		"sasl.mechanism":         "PLAIN",
		"sasl.username":          username,
		"sasl.password":          password,
		"heartbeat.interval.ms":  100,
		"max.poll.interval.ms":   6000,
		"session.timeout.ms":     6000,
		"go.logs.channel.enable": true,
		"auto.offset.reset":      "earliest"})

	if err != nil {
		fmt.Fprintf(os.Stderr, "Failed to create consumer: %s\n", err)
		os.Exit(1)
	}

	fmt.Printf("Created Consumer %v\n", c)

	if err != nil {
		fmt.Printf("Failed to create schema registry client: %s\n", err)
		os.Exit(1)
	}

	if err != nil {
		fmt.Printf("Failed to create deserializer: %s\n", err)
		os.Exit(1)
	}

	err = c.SubscribeTopics(topics, nil)
	if err != nil {
		panic(err)
	}

	go func() {
		for ev := range c.Logs() {
			fmt.Println("Kafka Logs: ", ev.String())
		}
	}()

	run := true

	for run {
		select {
		case sig := <-sigchan:
			fmt.Printf("Caught signal %v: terminating\n", sig)
			run = false
		default:
			ev := c.Poll(100)
			if ev == nil {
				continue
			}

			switch e := ev.(type) {
			case *kafka.Message:
				fmt.Println("Message Received")
				time.Sleep(10 * time.Second)
			case kafka.Error:
				// Errors should generally be considered
				// informational, the client will try to
				// automatically recover.
				fmt.Fprintf(os.Stderr, "%% Error: %v: %v\n", e.Code(), e)
			default:
				fmt.Printf("Ignored %v\n", e)
			}
		}
	}

	fmt.Printf("Closing consumer\n")
	c.Close()
}
```

## Logs with Channel Enabled

```
Created Consumer rdkafka#consumer-1
Message Received
Message Received
Message Received
Message Received
Message Received
Message Received
Message Received
Message Received
^CCaught signal interrupt: terminating
Closing consumer
```

Checklist
=========
Please provide the following information:

 - [x] confluent-kafka-go and librdkafka version (`LibraryVersion()`): v1.9.2
 - [x] Apache Kafka broker version: Using Confluent Managed Kafka
 - [x] Client configuration: `ConfigMap{...}`: Present in code
 - [x] Operating system: MacOS. Observed in Linux as well
 - [x] Provide client logs (with `"debug": ".."` as necessary)
 - [x] Provide broker log excerpts: can't provide as we are using managed kafka
 - [x] Critical issue

**Comment by milindl** [comment]
We're looking into this internally, it seems to be a librdkafka issue, rather than just a go binding issue. Thanks for reporting this!

**Comment by milindl** [comment]
Fixed with librdkafka #4176
This will make its way into the next release, closing this issue. thank you for reporting it.
